command-line/lib/Options/Applicative/Credential.hs

```diff
@@ -1,30 +1,48 @@
+{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
 
 {-# OPTIONS_HADDOCK hide #-}
 
 module Options.Applicative.Credential
-    (
-    -- * Types
-      CredentialType (..)
-
-    -- * Applicative Parser
-    , credentialOpt
+    ( delegationCredentialArg
     ) where
 
+import Prelude
+
+import Cardano.Address.Derivation
+    ( Depth (..) )
 import Cardano.Address.Style.Shelley
-    ( CredentialType (..) )
+    ( Credential (..), liftXPub )
 import Control.Applicative
     ( (<|>) )
 import Options.Applicative
-    ( Parser, flag', long )
+    ( Parser, argument, readerError )
+import Options.Applicative.Derivation
+    ( xpubArg )
+import Options.Applicative.Script
+    ( scriptHashArg )
+
+import qualified Cardano.Codec.Bech32.Prefixes as CIP5
 
 --
 -- Applicative Parser
 --
 
--- | Parse an 'CredentialType' from the command-line, as set of non-overlapping flags.
-credentialOpt :: Parser CredentialType
-credentialOpt = fromKeyFlag <|> fromScriptFlag
+delegationCredentialArg  :: String -> Parser (Credential 'DelegationK)
+delegationCredentialArg str = fromKey <|> fromScript <|> failure
   where
-    fromKeyFlag = flag' CredentialFromKey (long "from-key")
-    fromScriptFlag = flag' CredentialFromScript (long "from-script")
+    fromKey =
+        DelegationFromKey . liftXPub <$> xpubArg allowedPrefixes str
+      where
+        -- TODO: Allow non-extended keys here.
+        allowedPrefixes =
+            [ CIP5.stake_xvk
```

Comment:
don't we miss `addr_xvk` and `script` here? Ie., `stake_xvk` and `addr_xvk` should be prefixes for from key and `script` prefix should be for from script credential..

Reply:
The script is captured by the `fromScript` parser, which uses `scriptHashArg` from `Options.Applicative.Script`. As for `addr_xvk`, it's not allowed here because it's a delegation credential, so we expect either a stake key or a script.